RTK incompatible with CUDA 11

Hi,

I tried to compile RTK as a module of ITK 5.1.1 using CUDA Toolkit 11.1.0 on Windows 10 using CMake 3.17.4 and Visual Studio 2019 v16.7.5.
This fails with the following error:
```
>------ Erstellen gestartet: Projekt: RTK, Konfiguration: Release x64 ------
>Generating ../../itkFunctionBase.xml
>Building NVCC (Device) object Modules/Remote/RTK/src/CMakeFiles/cuda_compile_1.dir/Release/cuda_compile_1_generated_rtkCudaAverageOutOfROIImageFilter.cu.obj
>nvcc fatal   : Unsupported gpu architecture 'compute_30'
>CMake Error at cuda_compile_1_generated_rtkCudaAverageOutOfROIImageFilter.cu.obj.Release.cmake:220 (message):
>  Error generating
>  C:/[...]/ITK-5.1.1-bin/Modules/Remote/RTK/src/CMakeFiles/cuda_compile_1.dir//Release/cuda_compile_1_generated_rtkCudaAverageOutOfROIImageFilter.cu.obj
```

Looking at the corresponding ```NVCC_CUDA_FLAGS``` inherited from [here](https://github.com/SimonRit/RTK/blob/019b327d2fd26d97e61b43c4cd6167e04807b573/cmake/FindCUDA_wrap.cmake#L62):
```
set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS}
     -gencode arch=compute_30,code=sm_30
     -gencode arch=compute_35,code=sm_35
     -gencode arch=compute_35,code=compute_35
    )
````
I see that compute capability (CC) 3.0 is required at minimum.
However, the minimum CC for CUDA 11 is 3.5 as of [this link](https://stackoverflow.com/a/28933055) - I couldn't find any official/better statement, sorry.
I have not tried to change anything because I cannot foresee the effects.

Best,
Moritz

--- a/cmake/FindCUDA_wrap.cmake
+++ b/cmake/FindCUDA_wrap.cmake
@@ -42,28 +42,19 @@ if (CUDA_FOUND)
   cuda_include_directories (${CMAKE_CURRENT_SOURCE_DIR})
 endif ()
 
-if("${CUDA_VERSION}" VERSION_LESS 8.0)
+if("${CUDA_VERSION}" VERSION_LESS 9.0)
  set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS}
-     -gencode arch=compute_20,code=sm_20
-     -gencode arch=compute_30,code=sm_30
-     -gencode arch=compute_35,code=sm_35
-     -gencode arch=compute_35,code=compute_35
-     )
-elseif("${CUDA_VERSION}" VERSION_LESS 9.0)
- set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS}
-     -Wno-deprecated-gpu-targets
-     -gencode arch=compute_20,code=sm_20
-     -gencode arch=compute_30,code=sm_30
-     -gencode arch=compute_35,code=sm_35
-     -gencode arch=compute_35,code=compute_35
-     )
-else()
- set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS}
-     -gencode arch=compute_30,code=sm_30
-     -gencode arch=compute_35,code=sm_35
-     -gencode arch=compute_35,code=compute_35
-     )
+     -gencode arch=compute_20,code=sm_20)
 endif()
+if("${CUDA_VERSION}" VERSION_LESS 11.0)
+ set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS}
+     -gencode arch=compute_30,code=sm_30)
+endif()
+set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS}
+     -Wno-deprecated-gpu-targets
+     -gencode arch=compute_35,code=sm_35
+     -gencode arch=compute_35,code=compute_35
+     )
 
 if(NOT "-std=c++${CMAKE_CXX_STANDARD}" IN_LIST CUDA_NVCC_FLAGS)
   list(APPEND CUDA_NVCC_FLAGS "-std=c++${CMAKE_CXX_STANDARD}")

